### ISSUE TITLE
Module import case sensitivity

### ISSUE THREAD

**Issue by david-grogan** [microsoft/TypeScript#27943]
I was tearing my hair out with this one for a few days, finally found the problem...

I have the following modules: home.ts, data.ts and pager.ts; pager.ts exports a single class Pager, containing some static variables, these static vars were being set in data.ts on the initialization of my app, and then when home.ts created an instance of the Pager class the code in pager.ts tried to access these supposedly initialized statics only to find them ```undefined```.

This had been working fine until I upgraded to VS 15.8.6 last week.

The problem it turns out was I had used different cases in the import statements:

```
// data.ts
import { Pager } from "../controls/Pager";

// home.ts
import { Pager } from "../../controls/pager";

```

Presumably in the previous version TypeScript instantiated a single module, called Pager, in both cases; but in the new version I'm getting two different modules, one called 'pager' and the other 'Pager', the code in data.ts was updating the static variables in the latter, and the code home.ts was using the former module where these statics had never been initialized. If I had a casing issue with any of the symbol names then the compiler would give me an undeclared identifier error, but it just silently fails when it comes to the right-hand-side of the import statement. So it should either revert to the previous behavior or throw a compiler error on the invalid casing used in the path (obviously in terms of Windows, which I'm using, both are valid file names referring to the same file, so maybe this isn't the correct thing to do on Windows).

**Comment by j-oliveras** [comment]
You can use `forceConsistentCasingInFileNames` option.

Don't check that is the correct file name case with the file system, but check that all import that resolve to a file has the same case.

**Comment by david-grogan** [comment]
Adding '''"forceConsistentCasingInFileNames": true,''' to my tsconfig.json doesn't make any difference, both:
```
// data.ts
import { Pager } from "../controls/Pager";```
and
```// home.ts
import { Pager } from "../../controls/pager";``
still compile without errors.

Even if that worked though, it wouldn't explain what has changed since the previous version of TypeScript.

**Comment by paulo-fernando-silva** [comment]
I ran into this issue. I was getting the impression that it was impossible to have static members in TS, even given that typescriptlang states that [Static Properties](https://www.typescriptlang.org/docs/handbook/classes.html) are possible.

This behavior leads to very hard to debug errors and should not be possible. It would all be much easier if the transpiler reported an error and refused to work in such situations.

I attached a simple sample that I used to isolate the issue [StaticScopeTest.zip](https://github.com/Microsoft/TypeScript/files/2497765/StaticScopeTest.zip)
